src/components/store-review/ImprovementForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState } from 'react';
+import { ScrollView } from 'react-native';
+import { BottomSheetTextInput } from '@gorhom/bottom-sheet';
+import { Trans, t } from '@lingui/macro';
+import { Controller, useForm } from 'react-hook-form';
+
+import { FillButton, IconButton, Stack, Text } from '~components/uikit';
+import { styled } from '~styles';
+import { sleep } from '~utils/common';
+import { showToast } from '~components/common/Toaster';
+
+type ImprovementFormType = {
+  setImprovementRequest: (value: boolean) => void;
```

Comment:
Instead of passing the whole setter function it might be better to pass a restricted function like `onCancel` that does the `setImprovementRequest(false)` and also replace `setOpen` with something like `onDone` that does `setOpen(false)`.